client/src/next-customer/components/NextCustomer.tsx

```diff
@@ -0,0 +1,69 @@
+import { Button, Container, Row, Col } from "react-bootstrap";
+import { Service, Props } from "../../intefaces/types.ts";
+import "../next-customer.css";
+import { useDate } from "../hooks/clock.ts";
+import { useEffect, useState } from "react";
+import API from "../../API/API.ts";
+
+export default function NextCustomer(props: Props) {
+  // get serviceName from serviceID
+  const myService = props.services.find(
+    (service: Service) => service.ServiceID === props.ticket.serviceID
+  );
+  const { time } = useDate();
+
+  const [officerId, setOfficerId] = useState(1);
+  const [ticketNumber, setTicketNumber] = useState(props.ticket.ticketID);
+  const [serviceName, setServiceName] = useState<string | undefined>(undefined);
+  const [estimatedTime, setEstimatedTime] = useState(props.ticket.estimatedTime);
+
+  const handleButton = async () => {
+    const ticket = await API.nextCustomer(officerId);
+    setEstimatedTime(ticket.estimatedTime);
+    setTicketNumber(ticket.ticketID);
+    setServiceName(
+      props.services.find((service: Service) => service.ServiceID === ticket.serviceID)?.ServiceName
+    );
+  };
+
+  useEffect(() => {
+    setServiceName(myService?.ServiceName);
+    console.log("Service Name: ", serviceName);
+  }, [myService, serviceName]);
```

Comment:
These useEffect needs to run only when myService changes. 

Remove the serviceName to the dependency array because it causes an unnecessary re-render loop, as setServiceName triggers the effect again.

Reply:
Done